src/mpi_t/mpit_initthread.c

```diff
@@ -5,6 +5,52 @@
 
 #include "mpiimpl.h"
 
+/*
+ * Read global config files that have MPICH_ environment variables for us to
+ * use
+ */
+static int read_config_files(const char *fname)
+{
+    int lineno = 0;
+    char storage[1024], *line;
+    FILE *f;
+
+    if (!(f = fopen(fname, "r"))) {
+        return FALSE;
+    }
+
+    while ((line = fgets(storage, sizeof(storage), f))) {
+        char key[128], val[512], *spot;
+        bool force = false, already;
+
+        lineno++;
+
+        if ('\n' == line[0] || '#' == line[0])
+            continue;
+        if ('\0' == line[0])
+            break;
+
+        if (2 > sscanf(line, "%127[^=]=%511s", key, val)) {
+            fprintf(stderr, "Error parsing config file %s line %d: %s\n", fname, lineno, line);
+            goto err;
+        }
+        /* If KEY:force=VALUE, then write prior values */
+        if ((spot = strstr(key, ":force"))) {
+            force = true;
+            *spot = '\0';
+        }
+
+        already = getenv(key);
```

Comment:
already isn't used in this patch. In my original patch, I used it to determine whether we should print some debugging info.

Reply:
I'm going to be out the next three weeks. If you cleaned this up I'd approve it.

Reply:
Done.